#summary Everything for WEAV-CLUSTERING

#sidebar TableOfContents

== Introduction ==

WEAV-CLUSTERING tries to divide a big read set into smaller read set with keeping following features as possible as we can.
  * A clustered read set has not so many genes together. We hope a clustered read set has only one gene.
  * All the isoform of a gene would be in a clustered read set if exist.

||*Definition* Cluster: we will call *a clustered read set* as *a cluster*.||

== Algorithm ==

Although we want to keep the above features during clustering, fortunately, this clustering step does not have to be perfect, because the assembly algorithm of WEAV can assemble multiple genes simultaneously.
Thus we propose a naive clustering algorithm as follows.
  # We construct a de Bruijn graph with K-tuples using the whole read set. In this graph, the edges are K-tuples, the nodes are (K-1)-tuples, and the weight of an edge is the number of the K-tuple in the read set.
  # We remove edges with very low weight (i.e., $\leq 1) to eliminate random edges caused by sequencing errors,
and we also remove edges with very high weight (i.e., $\geq$ 300) to eliminate edges caused by repetitive
sequences. (3) After edge removal, the graph is fragmented into many disjoint connected components, or
subgraphs, which can be identified using the DFS (depth-first-search) algorithm. (4) For
each subgraph, we extract reads from the $K$-tuple edges and group them into a cluster.
Then, a de Bruijn graph algorithm is applied to each cluster to assemble gene transcripts.
After clustering, the orientation of each read within a cluster is known.
Thus, in the assembly algorithm, we do not have to worry about the orientation of reads.


== Details ==
Add your content here.  Format your content with:
  * Text in *bold* or _italic_
  * Headings, paragraphs, and lists
  * Automatic links to other wiki pages